lib/Mojo/WebSocketProxy/Dispatcher.pm

```diff
@@ -66,7 +66,10 @@ sub open_connection {
         # We normalize the entire string before decoding.
         my $normalized_msg = Unicode::Normalize::NFC($msg);
         my $args = eval { decode_json_utf8($normalized_msg) };
-        $log->error("JSON decoding $normalized_msg failed: $@") if $@;
+        if ($@) {
+            $log->debug("JSON decoding failed for: $normalized_msg");
+            $log->error("JSON decoding error: $@") if $log->is_level('debug');
```

Comment:
shouldn't this be also a debug message instead of checking debug level?

Reply:
Originally the idea here is that we emit an error on the log only if the log level is debug, but I agree with @binarytom to just simplify this and emit to the debug log instead, foregoing any log level checks.